app.coffee

```diff
@@ -22,6 +22,7 @@ app.post '/user/:user_id/tag', controller.createTag
 app.post '/user/:user_id/tag/:tag_id/rename', controller.renameTag
 app.del  '/user/:user_id/tag/:tag_id', controller.deleteTag
 app.post '/user/:user_id/tag/:tag_id/project/:project_id', controller.addProjectToTag
+app.post '/user/:user_id/tag/name/:tag_name/project/:project_id', controller.addProjectToTagName
```

Comment:
Is the client going to block characters that are promblematic in URLs, like `/`? Have you considered passing the tag name in the body, so we can JSON encode and have a more robust encoding format?

Reply:
It should be automatically URL encoded by the http client right?

Reply:
I'm unsure, since it has to pass through web, which presumably builds the URL, since it needs to inject the `user_id`? I think as a general rule we tend to put ids in the URL, but any user content in the json body, to avoid needing worrying about this kind of thing.

Reply:
That makes sense, I will change it